src/supervisor/watchers/handlers/pagination.ts

```diff
@@ -99,3 +99,87 @@ export async function paginatedNamespacedList<
     body: list,
   };
 }
+
+/**
+ * This function ensures that when listing workloads from the Kubernetes API, they are paginated in batches of 100.
+ * The workloads collected are additionally trimmed to contain only the relevant data for vulnerability analysis.
+ * The combination of both listing and trimming ensures we reduce our memory footprint and prevent overloading the API server.
+ */
+export async function paginatedClusterList<
+  T extends KubernetesObject & Partial<{ status: unknown; spec: unknown }>,
```

Comment:
This function is almost identical to the paginatedNamespacedList function, can we make `namespace` parameter a optional parameter, and the reuse the paginatedNamespacedList function somehow? Unless we are planning on getting rid of it in the future?

Reply:
Unfortunately not possible to re-use because the order of parameters is different in these functions!

Reply:
you could wrap the params in an object?

Reply:
I don't think it's possible still to call the function, I cannot `await listFn(...params)`